src/marketplace/marketplace.page.jsx

```diff
@@ -25,6 +28,12 @@ const MarketplacePage = () => {
         }
     `;
 
+    const navToAsset= (event) => {
+        // event.data
+        const nft_id = event.target.parentElement.parentElement.id;
+        navigate('asset', { state: { id:nft_id }});

```

Comment:
We will be using did in the url instead not the did passed down through state
it easier to directly link Asset from anywhere or redirect users